test/spec/features/modeling/ToggleShapeCollapseSpec.js

```diff
@@ -158,6 +208,22 @@ describe('features/modeling - toggle collapsed', function() {
         })
       );
 
+
+      it('expand do not show children under collapsed parents',
+        inject(function(modeling, commandStack) {
+
+          // given
+          modeling.toggleCollapse(shownChildShape);
+          modeling.toggleCollapse(shapeToCollapse);
+
+          // when
+          commandStack.undo();
```

Comment:
This test case is fine, it does not cover the case [described here](https://github.com/bpmn-io/diagram-js/pull/421#issuecomment-543116802) though.

My previous comment in JS code:

```javascript
// given
modeling.toggleCollapse(shownChildShape);
modeling.toggleCollapse(shapeToCollapse);

// when
modeling.toggleCollapse(shapeToCollapse);

// then
// ?
```

Reply:
Yes, even though this test case has the same kind of behavior, I'll add another coverage for expanding instead of undoing. Besides that, in this test:
A -> shapeToCollapse
    B -> shownChildShape 
        C -> shownSubChildShape

We first collapse B, then collapse A, then undo which expands A and check if C is still invisible and B is still collapsed.

Reply:
I've created a failing test case regarding this: https://github.com/bpmn-io/diagram-js/pull/421/commits/2a9da7cffd92a0d1eb2c225b45d6c4ed7b432d6e

Reply:
Ah yes ok I see the mistake now. Included this case and made a fix. I initially considered this case then somehow unfixed this while refactoring. Thanks for the catch!